### PR TITLE
some misc userdomain fixes

### PR DESCRIPTION
**Pull request by etbe**

### CODE REVIEW

File: policy/modules/system/userdomain.if [hunk 1]
```diff
@@ -3667,6 +3670,25 @@ interface(`userdom_relabelfrom_user_runtime_dirs',`
...
+##     Domain allowed access.
+##     </summary>
+## </param>
+#
+interface(`userdom_write_all_user_runtime_named_sockets',`
+	gen_require(`
+		attribute user_runtime_content_type;
+	')
+
+	allow $1 user_runtime_content_type:dir list_dir_perms;
```
Comment (0xC0ncord) [inline policy/modules/system/userdomain.if:3688]: Is it possible to downgrade this access to `search_dir_perms`?

File: policy/modules/system/userdomain.if [hunk 2]
```diff
@@ -69,6 +69,9 @@ template(`userdom_base_user_template',`
 	dontaudit $1_t user_tty_device_t:chr_file ioctl;
 
 	kernel_read_kernel_sysctls($1_t)
+	kernel_read_crypto_sysctls($1_t)
+	kernel_read_fs_sysctls($1_t)
+	kernel_read_vm_overcommit_sysctl($1_t)
```
Comment (0xC0ncord) [inline policy/modules/system/userdomain.if:74]: What are these for?

Comment (etbe) [reply policy/modules/system/userdomain.if:74]: type=AVC msg=audit(1695241764.653:6170): avc:  granted  { read open } for  pid=285781 comm="http" path="/proc/sys/crypto/fips_enabled" dev="proc" ino=21350 scontext=etbe:user_r:user_t:s0-s0:c0.c1023 tcontext=system_u:object_r:sysctl_crypto_t:s0 tclass=file

The crypto sysctls are needed by some http/https programs among other things, seems like a lot of library code checks this.

Comment (etbe) [reply policy/modules/system/userdomain.if:74]: type=AVC msg=audit(1695242496.734:181): avc:  granted  { read open } for  pid=1932 comm=66757365206D61696E6C6F6F70 path="/proc/sys/fs/pipe-max-size" dev="proc" ino=27353 scontext=user_u:user_r:user_t:s0 tcontext=system_u:object_r:sysctl_fs_t:s0 tclass=file

This process is xdg-document-portal. Lots of programs check pipe-max-size.

type=AVC msg=audit(1695242498.119:183): avc:  granted  { read open } for  pid=1673 comm=5468726561642028706F6F6C656429 path="/proc/sys/vm/overcommit_memory" dev="proc" ino=29888 scontext=user_u:user_r:user_t:s0 tcontext=system_u:object_r:sysctl_vm_overcommit_t:s0 tclass=file

This process is KDE plasmashell. Lots of programs do checks like this.

I don't think there's a downside to allowing these to be read.

Comment (0xC0ncord) [reply policy/modules/system/userdomain.if:74]: Seems fine. Was just curious as I don't think I have seen these AVCs before.

File: policy/modules/apps/pulseaudio.te [hunk 3]
```diff
@@ -149,6 +149,7 @@ userdom_search_user_home_content(pulseaudio_t)
 userdom_manage_user_tmp_dirs(pulseaudio_t)
 userdom_manage_user_tmp_files(pulseaudio_t)
 userdom_manage_user_tmp_sockets(pulseaudio_t)
+userdom_write_all_user_runtime_named_sockets(pulseaudio_t)
```
Comment (pebenito) [inline policy/modules/apps/pulseaudio.te:152]: Wouldn't it be better to limit this to only unpriv users?

File: policy/modules/system/userdomain.if (hunk 1 above)
Comment (etbe) [reply policy/modules/system/userdomain.if:3688]: OK I'll change that for now. We can merge less privs and then decide later whether to add more.

File: policy/modules/apps/pulseaudio.te (hunk 3 above)
Comment (etbe) [reply policy/modules/apps/pulseaudio.te:152]: What would be a good way of doing that?

Comment (pebenito) [reply policy/modules/apps/pulseaudio.te:152]: Nevermind, it slipped my mind how users were set up.

**Review by pebenito — APPROVED** [review]

**Pull request merged**